nkms/crypto/keyring/keys/signing_keypair.py

```diff
@@ -0,0 +1,58 @@
+import msgpack
+from random import SystemRandom
+from py_ecc.secp256k1 import N, privtopub, ecdsa_raw_sign, ecdsa_raw_recover
+
+
+class SigningKeypair(object):
+    def __init__(self, privkey_bytes=None):
+        self.secure_rand = SystemRandom()
+        if privkey_bytes:
+            self.priv_key = privkey_bytes
+        else:
+            # Key generation is random([1, N - 1])
+            priv_number = self.secure_rand.randrange(1, N)
+            self.priv_key = priv_number.to_bytes(32, byteorder='big')
+        # Get the public component
+        self.pub_key = privtopub(self.priv_key)
+
+    def _vrs_msgpack_dump(self, v, r, s):
+        v_bytes = v.to_bytes(1, byteorder='big')
+        r_bytes = r.to_bytes(32, byteorder='big')
+        s_bytes = s.to_bytes(32, byteorder='big')
+        return msgpack.dumps((v_bytes, r_bytes, s_bytes))
```

Comment:
Or, I guess, let's see how serialized signatures look like in Ethereum?

Reply:
Definitely. I couldn't find a quick reference to the signatures, so I just went with what I thought was best for the moment.

I'm using msgpack here because it answers the byte reading problem we had before. If I didn't do this, I'd have to do something bytestring substring stuff for it, and this just seemed much simpler.

In this case, I can pack it as a tuple and unpack it as a list keeping the elements in order. It "just works" haha.

I completely agree that we should see how Ethereum serializes signatures.

Reply:
Bytestring substring (I guess that's what you need for Ethereum-way) should be also simple though. E.g. for parsing:
```
assert (len(sig) - 1) == bitsize * 2
v_bytes = sig[0]
r_bytes = sig[1: len(sig) // 2 + 1]
s_bytes = sig[len(sig) // 2 + 1:]
```
Although, not as expressive as with msgpack of course